src/truncate.jl

```diff
@@ -17,11 +17,20 @@ function Truncated(d::ContinuousUnivariateDistribution, l::Float64, u::Float64)
     lcdf = isinf(l) ? 0.0 : cdf(d, l)
     ucdf = isinf(u) ? 1.0 : cdf(d, u)
     tp = ucdf - lcdf
-    Truncated{typeof(d)}(d, l, u, lcdf, ucdf, tp, log(tp))
+    Truncated{typeof(d),Continuous}(d, l, u, lcdf, ucdf, tp, log(tp))
 end
 
 @compat Truncated(d::ContinuousUnivariateDistribution, l::Real, u::Real) = Truncated(d, Float64(l), Float64(u))
 
+function Truncated(d::DiscreteUnivariateDistribution, l::Float64, u::Float64)
+    l < u || error("lower bound should be less than upper bound.")
+    lcdf = isinf(l) ? 0.0 : cdf(d, l)
+    ucdf = isinf(u) ? 1.0 : cdf(d, u)
+    tp = ucdf - lcdf
+    Truncated{typeof(d),Discrete}(d, l, u, lcdf, ucdf, tp, log(tp))
+end
+
+@compat Truncated(d::DiscreteUnivariateDistribution, l::Real, u::Real) = Truncated(d, Float64(l), Float64(u))
```

Comment:
It should be possible to use parametric dispatch to use the same methods for both Continuous and Discrete